op-deployer/pkg/deployer/state/deploy_config.go

```diff
@@ -101,14 +101,24 @@ func CombineDeployConfig(intent *Intent, chainIntent *ChainIntent, state *State,
 		},
 	}
 
+	if chainState.StartBlock == nil {
+		// These are dummy variables - see below for rationale.
+		num := rpc.LatestBlockNumber
+		cfg.L1StartingBlockTag = &genesis.MarshalableRPCBlockNumberOrHash{
+			BlockNumber: &num,
+		}
+	} else {
+		startHash := chainState.StartBlock.Hash()
+		cfg.L1StartingBlockTag = &genesis.MarshalableRPCBlockNumberOrHash{
+			BlockHash: &startHash,
+		}
+	}
```

Comment:
Why do we need this conditional now, was this a bug?

Reply:
The l1starting block tag isn't used by the l2genesis script, but it might be used by third party tooling like the superchain registry so as a convenience I set it to the correct value if it's available. The tag isn't available during the l2genesis step since the start block is set during a future pipeline stage.